source/helpers/math.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Linear interpolation. Given a range and a value, returns the
+ * percentage position in the range for the value.
+ *
+ * @param min Lowest value in the range
+ * @param max Highest value in the range
+ * @param interpolation Interpolation amount from 0.0 to 1.0 (or beyond to get a value outside the range [min, max])
+ * @returns Interpolated value
+ */
+export function lerp(min: number, max: number, interpolation: number): number {
+	return min + ((max - min) * interpolation);
+}
+
+/**
+ * Inverse operation of lerp. Given a range and a value, returns the
+ * interpolation amount to get that value.
+ *
+ * @param min Lowest value in the range
+ * @param max Highest value in the range
+ * @param value Value in the range
+ * @returns Interpolation amount from 0.0 to 1.0 (or beyond if `x` is outside the range [min, max])
+ */
+export function invlerp(min: number, max: number, value: number): number {
+	return (value - min) / (max - min);
+}
+
+/**
+ * Create a function that calculates a GitHub heat index based on a list of numbers.
+ * The list is used to find the min and max of the range. The function can then
+ * calculate a heat index for a given value based on that range.
+ *
+ * GitHub has 10 heat values, and uses a lower heat index for a higher value. So the
+ * highest heat index that will be returned is 1, and the lowest is 10.
+ *
+ * @param numbers values to use to calculate the min and max of the heat range
+ * @returns function to calculate heat index of a given number
+ */
+export function createHeatIndexFunc(numbers: number[]): (value: number) => number {
```

Comment:
Maybe I am not understanding what this does. But why cant this return a map?

Reply:
What would the map look like? The input value can be anything within the min/max range. It's not a limited set of possibilities.

Reply:
Ok I _think_ I understand what you are doing.
Your assigning the heatmap relative to the download count of the page?

So if I have only 2 releases with them both having 5k Downloads what will happen?

Reply:
That's right. In that case both releases would have the lowest heat since it floors the heat value, so they would both show up as gray.

Oh you made me realize that line 24 would have a divide by zero in that case, let me fix that.

Reply:
I think that the heat map should be a set number not relative to whats on the page.

500k should be dark red regardless if there are any other downloads on the page.

Reply:
Totally doable by passing in hardcoded min/max values to the func. I think @fregante wanted it to be dynamic though.

Reply:
@yakov116 the problem with hardcoded values is that most repos will appear just black. Making it dynamic would let us use the heatmap in place of "making the most downloaded file bolder"

Reply:
Ok 👍

Reply:
Any other revisions? Would like to get this in for hacktoberfest :D